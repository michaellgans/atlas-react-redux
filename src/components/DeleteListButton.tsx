// DeleteListButton Component

// Asset Imports
import { deleteList } from "../slices/listsSlice";
import { useAppDispatch } from "../store";

// Types
type DeleteListButtonProps = {
  listID: string;
};

// Returns a DeleteListButton Component
export function DeleteListButton({ listID }: DeleteListButtonProps) {
  // Define Hook
  const dispatch = useAppDispatch();

  const handleDeleteList = () => {
    dispatch(deleteList(listID));
  };

  return (
    <button className="h-[30px]" onClick={handleDeleteList}>
      <svg
        className="hidden h-[30px] w-[30px] cursor-pointer group-hover/list:block"
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth="{1.5}"
        stroke="currentColor"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          d="m9.75 9.75 4.5 4.5m0-4.5-4.5 4.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
        />
      </svg>
    </button>
  );
}
