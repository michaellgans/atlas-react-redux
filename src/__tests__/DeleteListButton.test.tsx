// Test for DeleteListButton Component

// Imports
import { render } from "@testing-library/react";
import { expect, test, it } from 'vitest';
import { DeleteListButton } from "../components/DeleteListButton.tsx";

it('Controll Test', () => {
    expect(2 + 2).toBe(4);
});

test('DeleteListButton renders correctly', () => {
    const { container } = render(<DeleteListButton />);
    expect(container).toMatchSnapshot();
});
