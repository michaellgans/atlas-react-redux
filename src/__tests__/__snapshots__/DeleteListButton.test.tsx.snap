// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteListButton renders correctly 1`] = `
<div>
  <button
    class="h-[30px]"
  >
    <svg
      class="hidden h-[30px] w-[30px] cursor-pointer group-hover/list:block"
      fill="none"
      stroke="currentColor"
      stroke-width="{1.5}"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m9.75 9.75 4.5 4.5m0-4.5-4.5 4.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
